src/GridFieldOrderableRows.php

```diff
@@ -531,24 +537,21 @@ protected function reorderItems($list, array $values, array $sortedIDs)
 
         // If not a ManyManyList and using versioning, detect it.
         $this->validateSortField($list);
-        $isVersioned = false;
         $class = $list->dataClass();
-
-        if (DataObject::getSchema()->tableName($class) == $this->getSortTable($list)) {
-            $isVersioned = $class::has_extension(Versioned::class);
-        }
+        $isVersioned = $class::has_extension(Versioned::class);
 
         // Loop through each item, and update the sort values which do not
         // match to order the objects.
-        if (!$isVersioned) {
+        if (!$isVersioned || $list instanceof ManyManyList) {
             $sortTable = $this->getSortTable($list);
-            $now = DBDatetime::now()->Rfc2822();
             $additionalSQL = '';
-            $baseTable = DataObject::getSchema()->baseDataTable($list->dataClass());
-
+            $baseTable = $sortTable;
+            if (class_exists($sortTable)) {
+                $baseTable = singleton($sortTable)->baseTable();
+            }
             $isBaseTable = ($baseTable == $sortTable);
             if (!$list instanceof ManyManyList && $isBaseTable) {
-                $additionalSQL = ", \"LastEdited\" = '$now'";
+                $additionalSQL = ', "LastEdited" = NOW()';
```

Comment:
Why have you moved from using `$now` to `NOW()`, this is a regression in behaviour...